### PR TITLE
Fix: bmqp messageproperties Wconv warns

### PR DESCRIPTION
**Pull request by melvinhe**
*Issue number of the reported bug or feature request: #87 *

**Describe your changes**
This PR is straightforward, addressing several ‘-Wconversion‘ errors relating to ‘bmqp_messageproperties.cpp‘ as well as associated header and test files with appropriate casting.

**Testing performed**
Double-checked that there's no data loss or overflow for each update. We had either checked sizes with assertions beforehand or the term was originally constrained to be of a relatively small size.

### CODE REVIEW

**Comment by melvinhe** [comment]
Still waiting for the CI checks to pass to know for sure, but these files are also highly-referenced so I expect these changes to eventually remove a thousand or so occurrences of '-Wconversion' in Build [ubuntu]. With the Build logs errs down to only 3,682 after #257 gets merged, this means we should be pretty close to clearing issue #87 .

**Review by pniedzielski — APPROVED** [review]
👍🏻

**Pull request merged**
